pkg/apis/core/validation/validation.go

```diff
@@ -6871,6 +6875,23 @@ func ValidateLoadBalancerStatus(status *core.LoadBalancerStatus, fldPath *field.
 				allErrs = append(allErrs, field.Invalid(idxPath.Child("ip"), ingress.IP, "must be a valid IP address"))
 			}
 		}
+
+		if utilfeature.DefaultFeatureGate.Enabled(features.LoadBalancerIPMode) {
+			if len(ingress.IP) > 0 && ingress.IPMode == nil {
+				allErrs = append(allErrs, field.Required(idxPath.Child("ipMode"), "must be specified when `ip` is set"))
```

Comment:
Why the backticks here? Error messages aren't Markdown.

Reply:
AFAIK status is only updated

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status

> When a new version of an object is POSTed or PUT, the spec is updated and available immediately. Over time the system will work to bring the status into line with the spec. 

but I do not think we should default, do we? https://github.com/kubernetes/kubernetes/pull/118895/files#r1247552003

@thockin @liggitt

Reply:
> I believe defaulting only occurs when creating new objects

Nope!  it magically happens any time a versioned object is created, including upon reading from storage.  Occasionally this is bad, but mostly it is good for us.  In this case it is good!  The `defaultOnRead()` stuff was added because:

```
// defaultOnRead sets interlinked fields that were not previously set on read.
// We can't do this in the normal defaulting path because that same logic
// applies on Get, Create, and Update, but we need to distinguish between them.
```

> Why the backticks here?

We wrote this down as a convention a looooooong time back, in an effort to make error messages clearer.  It's not to render markdown, but to denote "this is a field name".  I'm happy to revisit the convention (seriously!) but we should fix all the places that do it.

Reply:
> We wrote this down as a convention a looooooong time back, in an effort to make error messages clearer. It's not to render markdown, but to denote "this is a field name". I'm happy to revisit the convention (seriously!) but we should fix all the places that do it.

Is that convention documented somewhere user visible? Would be nice.

I think changing it might be a KEP&hellip; - possible, but more than trivial effort.

Reply:
> Is that convention documented somewhere user visible?

Probably not - we never considered the error messages to be API and made no guarantees on them.  As such, I don't think a KEP would be needed, just a clearly better convention.

Reply:
x-ref https://github.com/kubernetes/kubernetes/issues/119452 and https://github.com/kubernetes/kubernetes/pull/119454

Reply:
@pacoxu has found some test failures that appear to be related to this (links above). I'm not sure what the solution is, but this specific validation appears to be backwards incompatible in practice.